lib/signatures.ts

```diff
@@ -3,35 +3,41 @@ import { SignatureHashAlgorithm, SignatureResult } from './types';
 import { getDubHashSignature } from './dubhash';
 import { getUhashSignature } from './uhash';
 import pMap = require('p-map');
+import { join } from 'path';
 
 const { readFile } = promises;
 
 export async function getSignaturesByAlgorithm(
+  baseDir: string,
```

Comment:
I think we can avoid the usage of baseDir here and keep everything as it was.
Then have a clean up on filePaths content, after all fileSignatures have been generated.
What do you think?

Reply:
What do you think of create a function e.g. `filterUserInfoFromFilePaths` (or any better name)?
The domain/purpose of generate getSignaturesByAlgorithm should keep as it's, and should not care at all about dealing with name exposure on filePaths.

Reply:
Bear in mind the later on we want to add the option to allow the users to decide if they want to completely hide or not the filePaths and have a function that follow the SRP principle would be of help as it will deal specifically with filePaths

Reply:
Bear in mind the later on we want to add the option to allow the users to decide if they want to completely hide or not the filePaths… Having a function that follow the SRP principle would be of help as it will deal specifically with filePaths

Reply:
SRS is a good practice, and we should follow that regardless. As mentioned, the decision outside this forum was to run benchmarks on both solutions before we decide what to merge.

Reply:
Going to resolve this, since it's related to the alternative (PR45)